tests/update_workflow.go

```diff
@@ -5367,3 +5367,52 @@ func (s *FunctionalSuite) TestUpdateWorkflow_NewSpeculativeWorkflowTask_QueryFai
   9 WorkflowExecutionUpdateCompleted
 `, events)
 }
+
+func (s *FunctionalSuite) TestUpdateWorkflow_AdmittedUpdatesAreSentToWorkerInOrderOfAdmission() {
+	// If our implementation is not in fact ordering updates correctly, then it may be ordering them
+	// non-deterministically. This number should be high enough that the false-negative rate of the test is low, but
+	// must not exceed our limit on number of in-flight updates. If we were picking a random ordering then the
+	// false-negative rate would be 1/(nUpdates!).
+	nUpdates := 20
+	s.testCluster.host.dcClient.OverrideValue(s.T(), dynamicconfig.WorkflowExecutionMaxInFlightUpdates, nUpdates)
+
+	tv := testvars.New(s.T().Name())
+	tv = s.startWorkflow(tv)
+	for i := 0; i < nUpdates; i++ {
+		updateId := fmt.Sprint(i)
+		go func() { _, _ = s.sendUpdate(tv, updateId) }()
+		for {
```

Comment:
I wonder if it's slightly better to use `s.Eventually(func() bool { ... }` here? It would express the intent slightly better, but is effectively the same (the only other benefit is that it has a timeout).

Reply:
Thanks, that's nicer!